typer/main.py

```diff
@@ -228,11 +228,11 @@ def get_command(typer_instance: Typer) -> click.Command:
         or len(typer_instance.registered_commands) > 1
     ):
         # Create a Group
-        click_command = get_group(typer_instance)
```

Comment:
I've renamed from `click_command` to `click_group` mainly because otherwise it would return `Incompatible types in assignment (expression has type "Command", variable has type "Group")`. Seems to be a scope issue?

Reply:
It's a type error! My understanding is that Group is more specific than Command, and this function returns a Command. So you'd probably want to cast the result of get_group.

Reply:
@juped Thank you. 

- Do you suggest making so that `main.get_group` returns a `click.Group` (I believe this is a requirement for `mypy` to run successfully)? By looking at the example code in the issue page, the user was calling `get_group` from their script, so I'm assuming for `mypy` to run successfully we need `main.get_group` to return a `click.Group`?

If yes, then I'm trying to figure out how to cast the result of `main.get_group_from_info` (which is what `main.get_group` is calling) from `click.Command` to `click.Group`. Is this how you do it in Python?

`click.Group(get_group_from_info(TyperInfo(typer_instance)))`

Thank you!

Reply:
This is at the type hinting level, where I believe it's `cast()`.

Reply:
I tried using `return cast(click.Group, group)` in `main.get_group` (this requires an import of `cast ` from the `typing` module), however when I run `mypy` on the sample script in the issue I get 
```
main.py:16: error: "Command" has no attribute "command"
Found 1 error in 1 file (checked 1 source file)
```
which isn't that helpful as line 16. Mmm...